src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import java.time.LocalDate;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parser is involved with parsing user commands in Duke.
+ */
+public class Parser {
+
+    /**
+     * Parses user input to create a DukeCommand
+     * @param input User input to be parsed.
+     * @return a DukeCommand that contains a Command and a String
+     * containing more details
+     */
+    public static DukeCommand parseCommand(String input) {
+        try {
+            String[] params = input.split(" ", 2);
+
+            if (input.equals("bye")) {
+                return new DukeCommand(Command.BYE, "");
+
+            } else if (params[0].equals("delete")) {
+                return new DukeCommand(Command.DELETE, params[1]);
+
+            } else if (input.equals("list")) {
+                return new DukeCommand(Command.LIST, "");
+
+            } else if (params[0].equals("done")) {
+                return new DukeCommand(Command.DONE, params[1]);
+
+            } else if (params[0].equals("todo")) {
+                return new DukeCommand(Command.TODO, params[1]);
+
+            } else if (params[0].equals("deadline")) {
+                return new DukeCommand(Command.DEADLINE, params[1]);
+
+            } else if (params[0].equals("event")) {
+                return new DukeCommand(Command.EVENT, params[1]);
+
+            } else if (params[0].equals("find")) {
+                return new DukeCommand(Command.FIND, params[1]);
+            } else {
+                return new DukeCommand(Command.INVALID, "");
```

Comment:
Would a switch case here make the code more readable? I don't think its in the coding standard so it might just be a personal opinion but the block of code here looks like it fits exactly what a switch case would do.